packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -239,11 +239,35 @@ export class InternalModelManager {
   constructor(
     private readonly apiIdentifier: string,
     readonly connection: GadgetConnection,
-    readonly options?: { pluralApiIdentifier: string }
+    readonly options?: { pluralApiIdentifier: string; hasAmbiguousIdentifiers: boolean }
   ) {
     this.capitalizedApiIdentifier = camelize(apiIdentifier);
   }
 
+  private validateRecord(record: RecordData) {
+    if (!this.options || !this.options.hasAmbiguousIdentifiers) {
+      return true;
+    }
+    const keys = Object.keys(record);
+
+    // any keys that aren't the apiIdentifier are invalid
+    return !keys.some((key) => key !== this.apiIdentifier);
+  }
+
+  private getRecordFromData(record: RecordData, functionName: string) {
+    let recordData = record;
+
+    if (!this.validateRecord(record)) {
+      throw Error(`Invalid arguments found in variables. Did you mean to use ${functionName}({ ${this.apiIdentifier}: { ... } })?`);
```

Comment:
Probably best to throw a Gadget-y error, eh?

https://github.com/gadget-inc/js-clients/blob/main/packages/api-client-core/src/support.ts